# Ecommerce Purchases

I use Fake Data about some purchases done through Amazon!

all the data is fake and made-up.

**Import pandas and read in the Ecommerce Purchases csv file and set it to a DataFrame called ecom.**

In [1]:
import numpy as np
import pandas as pd

In [3]:
ecom = pd.read_csv("Ecommerce Purchases")
ecom

Address    Lot AM or PM  \
0     16629 Pace Camp Apt. 448\nAlexisborough, NE 77...  46 in       PM   
1     9374 Jasmine Spurs Suite 508\nSouth John, TN 8...  28 rn       PM   
2                      Unit 0065 Box 5052\nDPO AP 27450  94 vE       PM   
3                 7780 Julia Fords\nNew Stacy, WA 45798  36 vm       PM   
4     23012 Munoz Drive Suite 337\nNew Cynthia, TX 5...  20 IE       AM   
...                                                 ...    ...      ...   
9995      966 Castaneda Locks\nWest Juliafurt, CO 96415  92 XI       PM   
9996  832 Curtis Dam Suite 785\nNorth Edwardburgh, T...  41 JY       AM   
9997              Unit 4434 Box 6343\nDPO AE 28026-0283  74 Zh       AM   
9998               0096 English Rest\nRoystad, IA 12457  74 cL       PM   
9999     40674 Barrett Stravenue\nGrimesville, WI 79682  64 Hr       AM   

                                           Browser Info  \
0     Opera/9.56.(X11; Linux x86_64; sl-SI) Presto/2...   
1     Opera/8.93.(Windows 98; Win 9x 4.90; en-US) Pr...   
2     Mozilla/5.0 (compatible; MSIE 9.0; Windows NT ...   
3     Mozilla/5.0 (Macintosh; Intel Mac OS X 10_8_0 ...   
4     Opera/9.58.(X11; Linux x86_64; it-IT) Presto/2...   
...                                                 ...   
9995  Mozilla/5.0 (Windows NT 5.1) AppleWebKit/5352 ...   
9996  Mozilla/5.0 (compatible; MSIE 9.0; Windows NT ...   
9997  Mozilla/5.0 (Macintosh; U; Intel Mac OS X 10_7...   
9998  Mozilla/5.0 (Macintosh; Intel Mac OS X 10_8_8;...   
9999  Mozilla/5.0 (X11; Linux i686; rv:1.9.5.20) Gec...   

                              Company       Credit Card CC Exp Date  \
0                     Martinez-Herman  6011929061123406       02/20   
1     Fletcher, Richards and Whitaker  3337758169645356       11/18   
2          Simpson, Williams and Pham      675957666125       08/19   
3     Williams, Marshall and Buchanan  6011578504430710       02/24   
4           Brown, Watson and Andrews  6011456623207998       10/25   
...                               ...               ...         ...   
9995                    Randall-Sloan   342945015358701       03/22   
9996         Hale, Collins and Wilson   210033169205009       07/25   
9997                     Anderson Ltd  6011539787356311       05/21   
9998                         Cook Inc   180003348082930       11/17   
9999                       Greene Inc  4139972901927273       02/19   

      CC Security Code                  CC Provider  \
0                  900                 JCB 16 digit   
1                  561                   Mastercard   
2                  699                 JCB 16 digit   
3                  384                     Discover   
4                  678  Diners Club / Carte Blanche   
...                ...                          ...   
9995               838                 JCB 15 digit   
9996               207                 JCB 16 digit   
9997                 1                VISA 16 digit   
9998               987             American Express   
9999               302                 JCB 15 digit   

                               Email                                     Job  \
0                  pdunlap@yahoo.com  Scientist, product/process development   
1                 anthony41@reed.com                       Drilling engineer   
2     amymiller@morales-harrison.com                Customer service manager   
3        brent16@olson-robinson.info                       Drilling engineer   
4        christopherwright@gmail.com                             Fine artist   
...                              ...                                     ...   
9995          iscott@wade-garner.com                              Printmaker   
9996              mary85@hotmail.com                         Energy engineer   
9997               tyler16@gmail.com                      Veterinary surgeon   
9998         elizabethmoore@reid.net                Local government officer   
9999           rachelford@vaughn.com        

**Check the head of the DataFrame.**

In [4]:
ecom.head()

Address    Lot AM or PM  \
0  16629 Pace Camp Apt. 448\nAlexisborough, NE 77...  46 in       PM   
1  9374 Jasmine Spurs Suite 508\nSouth John, TN 8...  28 rn       PM   
2                   Unit 0065 Box 5052\nDPO AP 27450  94 vE       PM   
3              7780 Julia Fords\nNew Stacy, WA 45798  36 vm       PM   
4  23012 Munoz Drive Suite 337\nNew Cynthia, TX 5...  20 IE       AM   

                                        Browser Info  \
0  Opera/9.56.(X11; Linux x86_64; sl-SI) Presto/2...   
1  Opera/8.93.(Windows 98; Win 9x 4.90; en-US) Pr...   
2  Mozilla/5.0 (compatible; MSIE 9.0; Windows NT ...   
3  Mozilla/5.0 (Macintosh; Intel Mac OS X 10_8_0 ...   
4  Opera/9.58.(X11; Linux x86_64; it-IT) Presto/2...   

                           Company       Credit Card CC Exp Date  \
0                  Martinez-Herman  6011929061123406       02/20   
1  Fletcher, Richards and Whitaker  3337758169645356       11/18   
2       Simpson, Williams and Pham      675957666125       08/19   
3  Williams, Marshall and Buchanan  6011578504430710       02/24   
4        Brown, Watson and Andrews  6011456623207998       10/25   

   CC Security Code                  CC Provider  \
0               900                 JCB 16 digit   
1               561                   Mastercard   
2               699                 JCB 16 digit   
3               384                     Discover   
4               678  Diners Club / Carte Blanche   

                            Email                                     Job  \
0               pdunlap@yahoo.com  Scientist, product/process development   
1              anthony41@reed.com                       Drilling engineer   
2  amymiller@morales-harrison.com                Customer service manager   
3     brent16@olson-robinson.info                       Drilling engineer   
4     christopherwright@gmail.com                             Fine artist   

        IP Address Language  Purchase Price  
0  149.146.147.205       el           98.14  
1     15.160.41.51       fr           70.73  
2   132.207.160.22       de            0.95  
3     30.250.74.19       es           78.04  
4     24.140.33.94       es           77.82

**How many rows and columns are there?**

In [5]:
ecom.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10000 entries, 0 to 9999
Data columns (total 14 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   Address           10000 non-null  object 
 1   Lot               10000 non-null  object 
 2   AM or PM          10000 non-null  object 
 3   Browser Info      10000 non-null  object 
 4   Company           10000 non-null  object 
 5   Credit Card       10000 non-null  int64  
 6   CC Exp Date       10000 non-null  object 
 7   CC Security Code  10000 non-null  int64  
 8   CC Provider       10000 non-null  object 
 9   Email             10000 non-null  object 
 10  Job               10000 non-null  object 
 11  IP Address        10000 non-null  object 
 12  Language          10000 non-null  object 
 13  Purchase Price    10000 non-null  float64
dtypes: float64(1), int64(2), object(11)
memory usage: 1.1+ MB


## What is the average Purchase Price?

In [6]:
ecom['Purchase Price'].mean()

50.34730200000025

> The average Purchase Price is **$50.347**

## What were the highest and lowest purchase prices?

In [7]:
ecom['Purchase Price'].max()

99.99

In [8]:
ecom['Purchase Price'].min()

0.0

> The highest is **99.99 dollar** The lowest is **0 dollar**

## How many people have English 'en' as their Language of choice on the website?

In [9]:
ecom["Language"].unique()

array(['el', 'fr', 'de', 'es', 'ru', 'pt', 'zh', 'en', 'it'], dtype=object)

In [14]:
len(ecom[ecom["Language"] == "en"])

1098

In [19]:
ecom[ecom["Language"] == "en"].value_counts()

Address                                                  Lot    AM or PM  Browser Info                                                                                                                          Company                    Credit Card       CC Exp Date  CC Security Code  CC Provider    Email                     Job                                   IP Address       Language  Purchase Price
000 Jared Shoals Apt. 244\nRamirezmouth, IL 12450        55 Zz  PM        Mozilla/5.0 (compatible; MSIE 6.0; Windows CE; Trident/3.1)                                                                           Garcia, Sanchez and Fox    3096295094562282  02/17        657               Discover       beltrancarmen@rogers.org  Biomedical engineer                   91.169.108.79    en        80.20             1
7434 Monica Rapids\nSouth Ryan, MT 24925                 43 xr  PM        Mozilla/5.0 (compatible; MSIE 6.0; Windows 98; Trident/3.1)                                                    

In [15]:
ecom[ecom["Language"] == "en"].count()

Address             1098
Lot                 1098
AM or PM            1098
Browser Info        1098
Company             1098
Credit Card         1098
CC Exp Date         1098
CC Security Code    1098
CC Provider         1098
Email               1098
Job                 1098
IP Address          1098
Language            1098
Purchase Price      1098
dtype: int64

> The number of users who have selected English 'en' as their preferred language on the website is **1098**

## How many people have the job title of "Lawyer" ? 


In [16]:
ecom[ecom["Job"] == "Lawyer"].info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 30 entries, 470 to 9979
Data columns (total 14 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   Address           30 non-null     object 
 1   Lot               30 non-null     object 
 2   AM or PM          30 non-null     object 
 3   Browser Info      30 non-null     object 
 4   Company           30 non-null     object 
 5   Credit Card       30 non-null     int64  
 6   CC Exp Date       30 non-null     object 
 7   CC Security Code  30 non-null     int64  
 8   CC Provider       30 non-null     object 
 9   Email             30 non-null     object 
 10  Job               30 non-null     object 
 11  IP Address        30 non-null     object 
 12  Language          30 non-null     object 
 13  Purchase Price    30 non-null     float64
dtypes: float64(1), int64(2), object(11)
memory usage: 3.5+ KB


> The number of individuals with the job title 'Lawyer' is **30**

## How many people made the purchase during the AM and how many people made the purchase during PM ?

**(Hint: Check out [value_counts()](http://pandas.pydata.org/pandas-docs/stable/generated/pandas.Series.value_counts.html) ) **

In [18]:
ecom["AM or PM"].value_counts()

PM    5068
AM    4932
Name: AM or PM, dtype: int64

> The number of people who made purchases during the AM is **4932**, and the number of people who made purchases during the PM is **5068**.

## What are the 5 most common Job Titles?

In [23]:
ecom["Job"].value_counts().sort_values(ascending=False).head(5)

Interior and spatial designer    31
Lawyer                           30
Social researcher                28
Purchasing manager               27
Designer, jewellery              27
Name: Job, dtype: int64

> **The 5 most common Job Titles**
> 1. Interior and spatial designer    31
> 2. Lawyer                           30
> 3. Social researcher                28
> 4. Purchasing manager               27
> 5. Designer, jewellery              27

## Someone made a purchase that came from Lot: "90 WT" , what was the Purchase Price for this transaction?

In [25]:
ecom[ecom["Lot"] == "90 WT"]["Purchase Price"]

513    75.1
Name: Purchase Price, dtype: float64

> The purchase price for the transaction that came from Lot '90 WT' is **$75.1**

## What is the email of the person with the following Credit Card Number: 4926535242672853

In [27]:
ecom[ecom["Credit Card"] == 4926535242672853]["Email"]

1234    bondellen@williams-garza.com
Name: Email, dtype: object

> The email associated with the credit card number 4926535242672853 is **bondellen@williams-garza.com**

## How many people have American Express as their Credit Card Provider *and* made a purchase above $95?

In [34]:
ecom[ecom["CC Provider"] == "American Express"][ecom["Purchase Price"] > 95].count()

C:\Users\pinos\AppData\Local\Temp\ipykernel_11608\853926099.py:1: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  ecom[ecom["CC Provider"] == "American Express"][ecom["Purchase Price"] > 95].count()


Address             39
Lot                 39
AM or PM            39
Browser Info        39
Company             39
Credit Card         39
CC Exp Date         39
CC Security Code    39
CC Provider         39
Email               39
Job                 39
IP Address          39
Language            39
Purchase Price      39
dtype: int64

In [74]:
ecom[(ecom['CC Provider']=='American Express') & (ecom['Purchase Price']>95)].count()

Address             39
Lot                 39
AM or PM            39
Browser Info        39
Company             39
Credit Card         39
CC Exp Date         39
CC Security Code    39
CC Provider         39
Email               39
Job                 39
IP Address          39
Language            39
Purchase Price      39
CC Exp Year         39
dtype: int64

> The number of individuals who have American Express as their Credit Card Provider and made a purchase above $95 is **39**

## How many people have a credit card that expires in 2025?

In [38]:
ecom2 = ecom
ecom2.head()

Address    Lot AM or PM  \
0  16629 Pace Camp Apt. 448\nAlexisborough, NE 77...  46 in       PM   
1  9374 Jasmine Spurs Suite 508\nSouth John, TN 8...  28 rn       PM   
2                   Unit 0065 Box 5052\nDPO AP 27450  94 vE       PM   
3              7780 Julia Fords\nNew Stacy, WA 45798  36 vm       PM   
4  23012 Munoz Drive Suite 337\nNew Cynthia, TX 5...  20 IE       AM   

                                        Browser Info  \
0  Opera/9.56.(X11; Linux x86_64; sl-SI) Presto/2...   
1  Opera/8.93.(Windows 98; Win 9x 4.90; en-US) Pr...   
2  Mozilla/5.0 (compatible; MSIE 9.0; Windows NT ...   
3  Mozilla/5.0 (Macintosh; Intel Mac OS X 10_8_0 ...   
4  Opera/9.58.(X11; Linux x86_64; it-IT) Presto/2...   

                           Company       Credit Card CC Exp Date  \
0                  Martinez-Herman  6011929061123406       02/20   
1  Fletcher, Richards and Whitaker  3337758169645356       11/18   
2       Simpson, Williams and Pham      675957666125       08/19   
3  Williams, Marshall and Buchanan  6011578504430710       02/24   
4        Brown, Watson and Andrews  6011456623207998       10/25   

   CC Security Code                  CC Provider  \
0               900                 JCB 16 digit   
1               561                   Mastercard   
2               699                 JCB 16 digit   
3               384                     Discover   
4               678  Diners Club / Carte Blanche   

                            Email                                     Job  \
0               pdunlap@yahoo.com  Scientist, product/process development   
1              anthony41@reed.com                       Drilling engineer   
2  amymiller@morales-harrison.com                Customer service manager   
3     brent16@olson-robinson.info                       Drilling engineer   
4     christopherwright@gmail.com                             Fine artist   

        IP Address Language  Purchase Price  
0  149.146.147.205       el           98.14  
1     15.160.41.51       fr           70.73  
2   132.207.160.22       de            0.95  
3     30.250.74.19       es           78.04  
4     24.140.33.94       es           77.82

In [39]:
ecom2.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10000 entries, 0 to 9999
Data columns (total 14 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   Address           10000 non-null  object 
 1   Lot               10000 non-null  object 
 2   AM or PM          10000 non-null  object 
 3   Browser Info      10000 non-null  object 
 4   Company           10000 non-null  object 
 5   Credit Card       10000 non-null  int64  
 6   CC Exp Date       10000 non-null  object 
 7   CC Security Code  10000 non-null  int64  
 8   CC Provider       10000 non-null  object 
 9   Email             10000 non-null  object 
 10  Job               10000 non-null  object 
 11  IP Address        10000 non-null  object 
 12  Language          10000 non-null  object 
 13  Purchase Price    10000 non-null  float64
dtypes: float64(1), int64(2), object(11)
memory usage: 1.1+ MB


In [44]:
ecom2["CC Exp Date"] = pd.to_datetime(ecom2["CC Exp Date"],format='%m/%y', errors='coerce')

In [46]:
ecom2.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10000 entries, 0 to 9999
Data columns (total 14 columns):
 #   Column            Non-Null Count  Dtype         
---  ------            --------------  -----         
 0   Address           10000 non-null  object        
 1   Lot               10000 non-null  object        
 2   AM or PM          10000 non-null  object        
 3   Browser Info      10000 non-null  object        
 4   Company           10000 non-null  object        
 5   Credit Card       10000 non-null  int64         
 6   CC Exp Date       10000 non-null  datetime64[ns]
 7   CC Security Code  10000 non-null  int64         
 8   CC Provider       10000 non-null  object        
 9   Email             10000 non-null  object        
 10  Job               10000 non-null  object        
 11  IP Address        10000 non-null  object        
 12  Language          10000 non-null  object        
 13  Purchase Price    10000 non-null  float64       
dtypes: datetime64[ns](1), f

In [50]:
ecom2["CC Exp Year"] = ecom2["CC Exp Date"].dt.year

In [51]:
ecom2.head()

Address    Lot AM or PM  \
0  16629 Pace Camp Apt. 448\nAlexisborough, NE 77...  46 in       PM   
1  9374 Jasmine Spurs Suite 508\nSouth John, TN 8...  28 rn       PM   
2                   Unit 0065 Box 5052\nDPO AP 27450  94 vE       PM   
3              7780 Julia Fords\nNew Stacy, WA 45798  36 vm       PM   
4  23012 Munoz Drive Suite 337\nNew Cynthia, TX 5...  20 IE       AM   

                                        Browser Info  \
0  Opera/9.56.(X11; Linux x86_64; sl-SI) Presto/2...   
1  Opera/8.93.(Windows 98; Win 9x 4.90; en-US) Pr...   
2  Mozilla/5.0 (compatible; MSIE 9.0; Windows NT ...   
3  Mozilla/5.0 (Macintosh; Intel Mac OS X 10_8_0 ...   
4  Opera/9.58.(X11; Linux x86_64; it-IT) Presto/2...   

                           Company       Credit Card CC Exp Date  \
0                  Martinez-Herman  6011929061123406  2020-02-01   
1  Fletcher, Richards and Whitaker  3337758169645356  2018-11-01   
2       Simpson, Williams and Pham      675957666125  2019-08-01   
3  Williams, Marshall and Buchanan  6011578504430710  2024-02-01   
4        Brown, Watson and Andrews  6011456623207998  2025-10-01   

   CC Security Code                  CC Provider  \
0               900                 JCB 16 digit   
1               561                   Mastercard   
2               699                 JCB 16 digit   
3               384                     Discover   
4               678  Diners Club / Carte Blanche   

                            Email                                     Job  \
0               pdunlap@yahoo.com  Scientist, product/process development   
1              anthony41@reed.com                       Drilling engineer   
2  amymiller@morales-harrison.com                Customer service manager   
3     brent16@olson-robinson.info                       Drilling engineer   
4     christopherwright@gmail.com                             Fine artist   

        IP Address Language  Purchase Price  CC Exp Year  
0  149.146.147.205       el           98.14         2020  
1     15.160.41.51       fr           70.73         2018  
2   132.207.160.22       de            0.95         2019  
3     30.250.74.19       es           78.04         2024  
4     24.140.33.94       es           77.82         2025

In [58]:
len(ecom2[ecom2["CC Exp Year"] == 2025])

1033

> The number of individuals with a credit card that expires in 2025 is **1033**

## What are the top 5 most popular email providers/hosts (e.g. gmail.com, yahoo.com, etc...)

In [63]:
ecom["Email"]

0                    pdunlap@yahoo.com
1                   anthony41@reed.com
2       amymiller@morales-harrison.com
3          brent16@olson-robinson.info
4          christopherwright@gmail.com
                     ...              
9995            iscott@wade-garner.com
9996                mary85@hotmail.com
9997                 tyler16@gmail.com
9998           elizabethmoore@reid.net
9999             rachelford@vaughn.com
Name: Email, Length: 10000, dtype: object

In [73]:
ecom['Email'].apply(lambda x: x.split('@')[1])

0                  yahoo.com
1                   reed.com
2       morales-harrison.com
3        olson-robinson.info
4                  gmail.com
                ...         
9995         wade-garner.com
9996             hotmail.com
9997               gmail.com
9998                reid.net
9999              vaughn.com
Name: Email, Length: 10000, dtype: object

In [71]:
ecom['Email'].apply(lambda x: x.split('@')[1]).value_counts().head()

hotmail.com     1638
yahoo.com       1616
gmail.com       1605
smith.com         42
williams.com      37
Name: Email, dtype: int64

> **The top 5 most popular email providers/hosts, based on user data, are**
> 1. hotmail.com     1638
> 2.yahoo.com       1616
> 3. gmail.com       1605
> 4. smith.com         42
> 5. williams.com      37

# Great Job!